examples/platform/silabs/SiWx917/SiWx917/sl_wifi_if.c

```diff
@@ -0,0 +1,813 @@
+    extra_info->beacon_rx_count   = test.beacon_rx_count - temp_reset->beacon_rx_count;
+    extra_info->mcast_rx_count    = test.mcast_rx_count - temp_reset->mcast_rx_count;
+    extra_info->mcast_tx_count    = test.mcast_tx_count - temp_reset->mcast_tx_count;
+    extra_info->ucast_rx_count    = test.ucast_rx_count - temp_reset->ucast_rx_count;
+    extra_info->ucast_tx_count    = test.ucast_tx_count - temp_reset->ucast_tx_count;
+    extra_info->overrun_count     = test.overrun_count - temp_reset->overrun_count;
+    return status;
+}
+
+/******************************************************************
+ * @fn   int32_t wfx_rsi_reset_count()
+ * @brief
+ *       Getting the driver reset count
+ * @param[in] None
+ * @return
+ *        status
+ *********************************************************************/
+int32_t wfx_rsi_reset_count()
+{
+    sl_wifi_statistics_t test = { 0 };
+    sl_status_t status        = SL_STATUS_OK;
+    status                    = sl_wifi_get_statistics(SL_WIFI_CLIENT_INTERFACE, &test);
+    VERIFY_STATUS_AND_RETURN(status);
+    temp_reset->beacon_lost_count = test.beacon_lost_count;
+    temp_reset->beacon_rx_count   = test.beacon_rx_count;
+    temp_reset->mcast_rx_count    = test.mcast_rx_count;
+    temp_reset->mcast_tx_count    = test.mcast_tx_count;
+    temp_reset->ucast_rx_count    = test.ucast_rx_count;
+    temp_reset->ucast_tx_count    = test.ucast_tx_count;
+    temp_reset->overrun_count     = test.overrun_count;
+    return status;
+}
+
+/******************************************************************
+ * @fn   wfx_rsi_disconnect()
+ * @brief
+ *       Getting the driver disconnect status
+ * @param[in] None
+ * @return
+ *        status
+ *********************************************************************/
+int32_t wfx_rsi_disconnect()
+{
+    return sl_wifi_disconnect(SL_WIFI_CLIENT_INTERFACE);
+}
+
+sl_status_t join_callback_handler(sl_wifi_event_t event, char * result, uint32_t result_length, void * arg)
+{
+    temp_reset = (wfx_wifi_scan_ext_t *) malloc(sizeof(wfx_wifi_scan_ext_t));
+    memset(temp_reset, 0, sizeof(wfx_wifi_scan_ext_t));
+    if (CHECK_IF_EVENT_FAILED(event))
+    {
+        SILABS_LOG("F: Join Event received with %u bytes payload\n", result_length);
+        callback_status = *(sl_status_t *) result;
+        wfx_rsi.join_retries += 1;
+        wfx_rsi.dev_state &= ~(WFX_RSI_ST_STA_CONNECTING | WFX_RSI_ST_STA_CONNECTED);
+        wfx_retry_interval_handler(is_wifi_disconnection_event, wfx_rsi.join_retries++);
+        if (is_wifi_disconnection_event || wfx_rsi.join_retries <= WFX_RSI_CONFIG_MAX_JOIN)
+        {
+            xEventGroupSetBits(wfx_rsi.events, WFX_EVT_STA_START_JOIN);
+        }
+        is_wifi_disconnection_event = true;
+        return SL_STATUS_FAIL;
+    }
+    /*
+     * Join was complete - Do the DHCP
+     */
+    SILABS_LOG("%s: join completed.", __func__);
```

Comment:
Unnecessary logs... This files should have been brought in in a completely different PR to alleviate the review process.

Reply:
Removed this file from this PR, adding it in a new PR